imsim/stamp.py

```diff
@@ -1,8 +1,54 @@
-
 from functools import lru_cache
+from dataclasses import dataclass, fields, MISSING
 import numpy as np
 import galsim
-from galsim.config import StampBuilder, RegisterStampType
+from galsim.config import StampBuilder, RegisterStampType, GetAllParams
+from lsst.obs.lsst.translators.lsst import SIMONYI_LOCATION as RUBIN_LOC
+from .opsim_meta import opsim_fallback_values
+
+from .diffraction_fft import apply_diffraction_psf
+from .readout import CcdReadout
+
+@dataclass
+class DiffractionPSF:
+    exptime: float
+    azimuth: galsim.Angle
+    altitude: galsim.Angle
+    rottelpos: galsim.Angle
+    psf_size: float = 4000
+    full_well: float = CcdReadout.full_well
+    pixel_length: float = 100.0  # TODO: fill in some reasonable value here.
+    latitude: galsim.Angle = RUBIN_LOC.lat
+    enabled: bool = True
+
+    def apply(self, image: galsim.Image) -> None:
+        if self.enabled:
+            apply_diffraction_psf(
+                image.array,
+                rottelpos=self.rottelpos.rad,
+                exptime=self.exptime,
+                latitude=self.latitude.rad,
+                azimuth=self.azimuth.rad,
+                altitude=self.altitude.rad,
```

Comment:
Here I am passing over to the lower level routines. This call acts as an interface to that layer. In lower level routines, I am always using radians so there, I dont need to check.
As a consequence, I dont have any overhead from the `Angle` wrapper and I also dont need to introduce a dependency to `galsim` / `coord` in the core routines.
With this in mind it would make life simpler if in the future this part of the code would be rewritten in a compiled language.

Do you still prefer keeping the `Angle` type here?

Reply:
Yes.  If you can document that there is noticeable overhead from using it, I would reconsider, but I'm pretty sure that there is essentially no overhead from the Angle class.

Reply:
It is not only, overhead. The `Angle` type is very invasive and I dont only use `np.cos` / `np.sin` on those values.
(E.g., I also use the `%` operator for the 4 fold symmetry, which is not supported for `Angle`).
Therefore at some point, I need to convert.